### PR TITLE
nufmt: init at 0.1.0

### PR DESCRIPTION
**Pull request by IogaMaster**
## Description of changes

Created package nufmt and added myself as a maintainer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IogaMaster** [comment]
If needed I will change this to `unstable-2023-09-25`

File: pkgs/development/tools/nufmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "nushell";
+    repo = "nufmt";
+    rev = "796ee834c1e31ead4c5479bf2827a4339c5d61d1";
+    hash = "sha256-BwKLl8eMCrqVt9PA5SHAXxu3ypP2ePcSuljKL+wSkvw=";
+  };
+
+  cargoSha256 = "sha256-JL1Ot2ldvieRM/SNLCC2YZ5b2hD98GXp+H4T/OFuKfk=";
```
Comment (Mic92) [inline pkgs/development/tools/nufmt/default.nix:18]: I got a different checksum here:

```
┏━ 2 Errors:
┃ error: builder for '/nix/store/v0jx1sgv5v0c6axjpcq87xmpgvalgspx-nufmt-0.1.0-vendor.tar.gz.drv' failed with exit code 1
┃ error: build of '/nix/store/v0jx1sgv5v0c6axjpcq87xmpgvalgspx-nufmt-0.1.0-vendor.tar.gz.drv' on 'ssh-ng://nix@ryan.dse.in.tum.de' failed: hash mismatch in fixed-output derivation '/nix/store/v0jx1sgv5v0c6axjpcq87xmpgvalgspx-nufmt-0.1.0-vendor.tar.gz.drv':
┃          specified: sha256-JL1Ot2ldvieRM/SNLCC2YZ5b2hD98GXp+H4T/OFuKfk=
┃             got:    sha256-SsOnuDnWkSRA5l8D04GqSwrAlijgWZNA8VdrZvB86SQ=
┣━ Dependency Graph:
┃       ┌─ ✔︎ source ⏱︎ 1s
┃    ┌─ ⚠︎ nufmt-0.1.0-vendor.tar.gz failed with exit code 1 after ⏱︎ 17s in fixupPhase
┃ ┌─ ⏳︎ nufmt-0.1.0
┃ ⏳︎ review-shell
┣━━━ Builds           │ Uploads   │ Host
┃        │ ✔︎ 1 │      │     │     │ localhost
┃        │     │      │     │ ↑︎ 1 │ ssh-ng://nix@ryan.dse.in.tum.de
┗━ ∑︎ ⏵︎ 0 │ ✔︎ 1 │ ⏳︎ 2 │ ↑︎ 0 │ ↑︎ 1 │ ⚠︎ Exited after 1 build failures at 20:45:15 after 25s

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/257297

1 package failed to build:
nufmt
```

**Comment by IogaMaster** [comment]
I'll change

File: pkgs/development/tools/nufmt/default.nix (hunk 1 above)
Comment (IogaMaster) [reply pkgs/development/tools/nufmt/default.nix:18]: Oh, i changed to use pname without thinking it would affect the hash, silly me

**Comment by IogaMaster** [comment]
Okay

**Comment by IogaMaster** [comment]
When can we merge?

**Comment by IogaMaster** [comment]
Is there anything else I need to do?

**Comment by IogaMaster** [comment]
https://github.com/nushell/nufmt/pull/41

**Comment by IogaMaster** [comment]
FYI I am working on a major rewrite of nufmt, although this would be fine to merge I'll update it in nixpkgs when it's time.
https://github.com/nushell/nufmt/pull/49

We now have a flake as well

**Comment by IogaMaster** [comment]
@Mic92 What do I need to do now?

**Comment by siph** [comment]
Result of `nixpkgs-review pr 257297` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nufmt</li>
  </ul>
</details>

**Comment by siph** [comment]
This builds and I was able to format an `.nu` file, but you should rename from `0.1.0` like you suggested earlier.

**Comment by IogaMaster** [comment]
@siph Okay, done

**Comment by siph** [comment]
You should also update/squish your commit message to `nufmt: init at unstable-2023-09-25`.

**Comment by IogaMaster** [comment]
Slight issue, there are many commits in between those. As I updated to keep in line with the master branch

**Comment by IogaMaster** [comment]
As I have made a mistake I am going to reopen this pr

**Comment by IogaMaster** [comment]
@siph 
New pr
https://github.com/NixOS/nixpkgs/pull/264500

**Pull request closed without merging**